### ISSUE TITLE
Can't run `@pika/plugin-build-types`

### ISSUE THREAD

**Issue by filips123** [pikapkg/builders#65]
I can't run `@pika/plugin-build-types` even if I have `typescript` installed. I tried to install `typescript` package both globally and locally, but it still doesn't work.

It looks like `@pika/plugin-build-types` [checks](https://github.com/pikapkg/builders/blob/master/packages/plugin-build-types/src/index.ts#L74) for `tsc` and `tsc.generateTypesForModule`. In my case, `tsc` exists but `tsc.generateTypesForModule` is `undefined`.

What is this? How to fix this?

Also, [error says](https://github.com/pikapkg/builders/blob/master/packages/plugin-build-types/src/index.ts#L85-L86) that I should install `typescript` package from `pika` command. What's that `pika` command? Should it be `yarn`? I installed `typescript` from yarn and as you can see it doesn't work.

This is command output of `pack build`:

```
[1/7] Validating source...
[2/7] Preparing pipeline...
      ❇️  pkg/
[3/7] Running @pika/plugin-standard-pkg...
      📝  pkg/dist-src/index.js [esnext]
      » Linting with standard-pkg...
[4/7] Running @pika/plugin-build-node...
      📝  pkg/dist-node/index.js [main]
[5/7] Running @pika/plugin-build-web...
      📝  pkg/dist-web/index.js [module]
[6/7] Running @pika/plugin-build-types...
      » no type definitions found, auto-generating...

⚠️  dist-types/: Attempted to generate type definitions, but "typescript" package was not found.
                Please install either locally or globally and try again.
       $ pika add --dev typescript
[alt.] $ pika global add typescript
[alt.] *   Write your own type definition file to "index.d.ts"

Error: Failed to build: dist-types/
    at /tmp/examples/simple-package/node_modules/@pika/plugin-build-types/dist-node/index.js:142:13
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/tmp/examples/simple-package/node_modules/@pika/plugin-build-types/dist-node/index.js:15:24)
    at _next (/tmp/examples/simple-package/node_modules/@pika/plugin-build-types/dist-node/index.js:37:9)
```

**Comment by FredKSchott** [comment]
🤦‍♂ that looks like a bad copy-paste on my part. That should be `yarn add typescript` or `npm install`

I'm not sure why `generateTypesForModule` is undefined, but I'm assuming you have an older TS version before that function was added? Can you share what version of TS you have installed in your project?

**Comment by filips123** [comment]
I have `typescript@3.5.3` installed as dev dependency with Yarn.

**Comment by FredKSchott** [comment]
That's really strange... looks like it was removed in TS 3.5. Trying to learn more about whether it was just moved/renamed or removed completely

**Comment by FredKSchott** [comment]
If that's the case (it appears to be so) then this plugin may not be worth it anymore. It would either need to be re-written to use dts-gen, or we could just pull it's remaining functionality (handling custom index.d.ts files) into pika/pack itself.

**Comment by filips123** [comment]
@FredKSchott Can it be rewritten to use `dts-gen`? Its code is very small so it shouldn't be hard to rewrite it.

It would be very good to keep existing functionality of generating types from jsdocs and code.

**Comment by FredKSchott** [comment]
I know, I would too. But if I had to guess what the TS team is thinking, it's: "if you're just going to auto-generate types from JS, then it would be better to just ship the JS and let the consumer do that auto-generation. That way, they always get the latest for their project vs. relying on the author to keep theirs up to date."

**Comment by FredKSchott** [comment]
Update here: https://twitter.com/43081j/status/1150883098567020545

I've downgraded the library to v3.4 where support still exists, so it should continue to work for now while we think up a better longterm strategy (ex: dts-gen)

**Comment by filips123** [comment]
@FredKSchott I now have a new problem:

```
$ pack build
@pika/pack build v0.4.0
[1/8] Validating source...
[2/8] Preparing pipeline...
      ❇️  pkg/
[3/8] Running @pika/plugin-standard-pkg...
      📝  pkg/dist-src/index.js [esnext]
      » Linting with standard-pkg...
[4/8] Running @pika/plugin-build-node...
      📝  pkg/dist-node/index.js [main]
[5/8] Running @pika/plugin-build-web...
      📝  pkg/dist-web/index.js [module]
[6/8] Running @pika/plugin-build-types...
      » no type definitions found, auto-generating...
TypeError: Cannot convert undefined or null to object
    at Function.getPrototypeOf (<anonymous>)
    at /usr/src/app/node_modules/typescript/lib/typescript.js:94725:53
    at /usr/src/app/node_modules/typescript/lib/typescript.js:94710:23
    at getValueInfo (/usr/src/app/node_modules/typescript/lib/typescript.js:94717:16)
    at /usr/src/app/node_modules/typescript/lib/typescript.js:94728:28
    at Object.flatMap (/usr/src/app/node_modules/typescript/lib/typescript.js:561:25)
    at /usr/src/app/node_modules/typescript/lib/typescript.js:94726:34
    at /usr/src/app/node_modules/typescript/lib/typescript.js:94710:23
    at getValueInfo (/usr/src/app/node_modules/typescript/lib/typescript.js:94717:16)
    at /usr/src/app/node_modules/typescript/lib/typescript.js:94728:28
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I now have `@pika/plugin-build-types@0.5.1` and `typescript@3.4.5` installed.

**Comment by FredKSchott** [comment]
Hmm, is there a repo I can run locally and check out?

**Comment by filips123** [comment]
Repo is currently not available, but I found out that it fails when `index.js` contains `export default MyClass` (or any other object).

**Comment by FredKSchott** [comment]
Futurama professor voice: Good news everyone, TypeScript v3.7 will come with a JS declaration generator! Does anyone want to take a stab at integrating? 

My guess is that it should be as simple as running `tsc --allowJs --declarations` on the `dist-web/index.js` file. You can probably grab most of the logic from our ts-standard-pkg plugin.

More info here: https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/

**Comment by FredKSchott** [comment]
Fixed in 2ba1f99
